kufar/books/main.go

```diff
@@ -38,6 +39,13 @@ type Book struct {
 	Author string
 	Date   time.Time
 	UserID int
+	Time   TimeStruct
+}
+
+//TimeStruct ...
+type TimeStruct struct {
+	sync.Mutex
```

Comment:
Please use `sync.RWMutex`. 
RW mutex allows to be locked on read or on write.
When you are making some writing stuff, you need to call `Lock()/Unlock()`. In reading stuff - `RLock()/RUnlock()`.
`A RWMutex is a reader/writer mutual exclusion lock. The lock can be held by an arbitrary number of readers or a single writer.`
[Doc](https://golang.org/pkg/sync/#RWMutex)